From 92c62c6728b8ad80b65d7e92fb727352dc82434e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 3 Nov 2023 14:35:54 +0100
Subject: [PATCH 0016/1695] core: kernel: interrupt: fix inline description

Fixes the inline description comment for type get_of_device_func
and API function dt_driver_device_from_parent().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d78975738575373ccd96ba9c85b434d096fc4bb1 core: kernel: interrupt: fix inline description]
Change-Id: I0f313497d74be9ad8ac991acc15d5df8f25385b8
Fixes: b357d34fe91f ("core: dt_driver: swap TEE_result and retrieved device reference")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from 1238110c075db550d08ccb213b2b00ddd0f929f2]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/include/kernel/dt_driver.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/core/include/kernel/dt_driver.h b/core/include/kernel/dt_driver.h
index 989be5a1a..d95f8b94c 100644
--- a/core/include/kernel/dt_driver.h
+++ b/core/include/kernel/dt_driver.h
@@ -102,13 +102,13 @@ struct dt_pargs {
  *
  * @parg: phandle argument(s) referencing the device in the FDT.
  * @data: driver private data registered in struct dt_driver.
- * @res: Output result code of the operation:
- *	TEE_SUCCESS in case of success
- *	TEE_ERROR_DEFER_DRIVER_INIT if device driver is not yet initialized
- *	Any TEE_Result compliant code in case of error.
+ * @out_device: output device reference upon success, e.g. a struct clk
+ *	pointer for a clock driver.
  *
- * Return a device opaque reference, e.g. a struct clk pointer for a clock
- * driver, or NULL if not found in which case @res provides the error code.
+ * Return code:
+ * TEE_SUCCESS in case of success
+ * TEE_ERROR_DEFER_DRIVER_INIT if device driver is not yet initialized
+ * Any TEE_Result compliant code in case of error.
  */
 typedef TEE_Result (*get_of_device_func)(struct dt_pargs *parg, void *data,
 					 void **out_device);
@@ -163,7 +163,7 @@ TEE_Result dt_driver_device_from_node_idx_prop(const char *prop_name,
  * @fdt: FDT base address
  * @nodeoffset: node offset in the FDT
  * @type: Driver type
- * @dout_device: output device opaque reference upon success, for example
+ * @out_device: output device opaque reference upon success, for example
  *	a struct i2c_dev pointer for a I2C bus driver
  *
  * Return code:
-- 
2.47.2

